test/unitTests/build/buildExecutor.test.ts

```diff
@@ -0,0 +1,304 @@
+import { EventStream } from '../../../src/common/eventStream';
+import { EnvironmentController } from '../../../src/common/environmentController';
+import { SinonSandbox, createSandbox, SinonStub } from 'sinon';
+import TestEventBus from '../../utils/testEventBus';
+import { getFakeEnvironmentController, getFakedTelemetryReporter, fakedExtensionContext, fakedBuildInput, testFolder, setTelemetryUserOptInToFalse, setTelemetryUserOptInToTrue, getFakedWindowsPlatformInformation, getFakedNonWindowsPlatformInformation } from '../../utils/faker';
+import { PlatformInformation } from '../../../src/common/platformInformation';
+import TelemetryReporter from '../../../src/telemetryReporter';
+import { BuildExecutor } from '../../../src/build/buildExecutor';
+import { DocfxRestoreStarted, DocfxRestoreCompleted, BuildCacheSizeCalculated, DocfxBuildStarted, DocfxBuildCompleted } from '../../../src/common/loggingEvents';
+import { DocfxExecutionResult, BuildResult } from '../../../src/build/buildResult';
+import { setTimeout } from 'timers';
+import { EventType } from '../../../src/common/eventType';
+import { Subscription } from 'rxjs';
+
+describe('BuildExecutor', () => {
+    let sinon: SinonSandbox;
+    let stubExecuteDocfx: SinonStub;
+
+    let eventStream: EventStream;
+    let subscription: Subscription;
+    let fakedEnvironmentController: EnvironmentController;
+    let fakedWindowsPlatformInformation: PlatformInformation;
+    let fakedNonWindowsPlatformInformation: PlatformInformation;
+    let fakedTelemetryReporter: TelemetryReporter;
+    let buildExecutor: BuildExecutor;
+    let testEventBus: TestEventBus;
+
+    let executedCommands: string[];
+    let executedOptions: cp.ExecOptions[];
+    let executedStdinInput: string[];
+    let killProcessTreeFuncCalled: boolean;
+
+    before(() => {
+        eventStream = new EventStream();
+        fakedEnvironmentController = getFakeEnvironmentController();
+        fakedWindowsPlatformInformation = getFakedWindowsPlatformInformation();
+        fakedNonWindowsPlatformInformation = getFakedNonWindowsPlatformInformation();
+        fakedTelemetryReporter = getFakedTelemetryReporter();
+        buildExecutor = new BuildExecutor(fakedExtensionContext, fakedWindowsPlatformInformation, fakedEnvironmentController, eventStream, fakedTelemetryReporter);
+
+        testEventBus = new TestEventBus(eventStream);
+
+        sinon = createSandbox();
+        stubExecuteDocfx = undefined;
+
+        sinon.stub(utils, 'getFolderSizeInMB').resolves(10);
+        sinon.stub(utils, 'killProcessTree').callsFake((pid: number, signal: string) => {
+            killProcessTreeFuncCalled = true;
+            return Promise.resolve();
+        });
+    });
+
+    function mockExecuteDocfx(restoreExitCode: number, buildExitCode: number) {
+        stubExecuteDocfx && stubExecuteDocfx.restore();
+        stubExecuteDocfx = sinon.stub(childProcessUtil, 'executeDocfx').callsFake(
+            function (command: string,

```

Comment:
I'm just not getting why stubbing `executeDocfx` here would make `buildExecutor` use the mocked `executeDocfx` function.

My hypothese is that once `buildExecutor` intance is created, the `executeDocfx` it imports from `childProcessUtils` should be immutable.s

Reply:
Hah, Cannot give you the answer now, not familiar with the implementation. 😯